File: kotlin-pr/50/kotlin/code-quality-error-handling/FileReader.kt
import java.io.File

class FileReader {
    fun readFile(filename: String): String {
        // {fact rule=code-quality-error-handling@v1.0 defects=1}
        return try {
            File(filename).readText()
        } catch (e: Exception) {
            "" // Hiding exception details by returning empty string
        }
        // {/fact}
    }
}

File: scala-pr/54/scala/code-quality-error-handling/FileReader.scala
import scala.io.Source

class FileReader {
  def readFile(filename: String): String = {
    // {fact rule=code-quality-error-handling@v1.0 defects=1}
    try {
      val source = Source.fromFile(filename)
      val content = source.mkString
      source.close()
      content
    } catch {
      case _: Exception => "" // Hiding exception details by returning empty string
    }
    // {/fact}
  }
}